lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -477,3 +505,31 @@ data ErrUpdateSealedTx
     -- key-witnesses would have been rendered invalid.
     | ErrByronTxNotSupported
     deriving (Generic, Eq, Show)
+
+-- | Error for when its impossible for 'distributeSurplus' to distribute the
+-- surplus. As long as the surplus is larger than 'costOfIncreasingCoin', this
+-- should never happen.
+newtype ErrMoreSurplusNeeded = ErrMoreSurplusNeeded Coin
+    deriving (Generic, Eq, Show)
+
+-- | Small helper record to disambiguate between a fee and change Coin values.
+-- Used by 'distributeSurplus'.
+data TxFeeAndChange change = TxFeeAndChange
+    { fee :: Coin
+    , change :: change
+    }
+    deriving (Eq, Show)
+
+-- | Manipulates a 'TxFeeAndChange' value.
+--
+txFeeAndChange
```

Comment:
I think it's worth calling this `mapTxFeeAndChange`. While `either` and `maybe` are well known, I don't think we use similar naming ourselves. E.g. `mapChainFollower`, `mapChainFollowerLog`

For brevity we could also have a separate `mapChange`, which would cover most of the use-cases in the implementation.

Reply:
Fair enough! Fixed in 1ba4983911176d6bbd78269ad0eae9dca60da6c9.

(I adjusted the original commit that added this function rather than making a new commit).